sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayAsyncClient.java

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.networktraversal;
+
+import com.azure.communication.networktraversal.implementation.CommunicationNetworkingClientImpl;
+import com.azure.communication.networktraversal.implementation.CommunicationNetworkTraversalsImpl;
+import com.azure.communication.networktraversal.models.CommunicationRelayConfiguration;
+import com.azure.communication.networktraversal.models.CommunicationErrorResponse;
+import com.azure.communication.networktraversal.models.CommunicationRelayConfigurationRequest;
+import com.azure.communication.networktraversal.models.CommunicationErrorResponseException;
+import com.azure.communication.common.CommunicationUserIdentifier;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import reactor.core.publisher.Mono;
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * Asynchronous client interface for Azure Communication Network Traversal
+ * operations
+ */
+@ServiceClient(builder = CommunicationRelayClientBuilder.class, isAsync = true)
+public final class CommunicationRelayAsyncClient {
+
+    private final CommunicationNetworkTraversalsImpl client;
+    private final ClientLogger logger = new ClientLogger(CommunicationRelayAsyncClient.class);
+
+    CommunicationRelayAsyncClient(CommunicationNetworkingClientImpl communicationNetworkingClient) {
+        client = communicationNetworkingClient.getCommunicationNetworkTraversals();
+    }
+
+    /**
+     * Creates a new CommunicationRelayConfiguration.
+     * 
+     * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
+     * @return The created Communication Relay Configuration.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CommunicationRelayConfiguration> getRelayConfiguration(CommunicationUserIdentifier communicationUser) {
+        CommunicationRelayConfigurationRequest body = new CommunicationRelayConfigurationRequest();
+        body.setId(communicationUser.getId());
+        
+        return Mono.just(client.issueRelayConfiguration(body));
```

Comment:
Do you need to wrap in Mono.just()? I think `client.issueRelayConfiguration(body)` returns Mono directly

Reply:
Yes, the one that is calling is of return type CommunicationRelayConfiguration in CommunicationTraversalImpl.java so Mono.just is needed. Also, I tested to in case this calling a different function.

Reply:
You should be calling `issueRelayConfigurationAsync`. `issueRelayConfiguration` is blocking and that is the opposite of what we want to do in an async client.

Reply:
Good call.